In [46]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

# Reading data(Super Market Sales)

In [47]:
df = pd.read_csv('supermarket_sales.csv')

# Exploring / Inspect the dataset

In [48]:
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4  
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3

In [49]:
df.Date.sort_values() #to solt label values

496    1/1/2019
484    1/1/2019
567    1/1/2019
523    1/1/2019
970    1/1/2019
         ...   
393    3/9/2019
73     3/9/2019
433    3/9/2019
585    3/9/2019
84     3/9/2019
Name: Date, Length: 1000, dtype: object

# To find null values

In [50]:
df.isnull().sum()

Invoice ID                 0
Branch                     0
City                       0
Customer type              0
Gender                     0
Product line               0
Unit price                 0
Quantity                   0
Tax 5%                     0
Total                      0
Date                       0
Time                       0
Payment                    0
cogs                       0
gross margin percentage    0
gross income               0
Rating                     0
dtype: int64

In [51]:
df.shape

(1000, 17)

In [52]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

In [53]:
df.describe()

Unit price     Quantity       Tax 5%        Total        cogs  \
count  1000.000000  1000.000000  1000.000000  1000.000000  1000.00000   
mean     55.672130     5.510000    15.379369   322.966749   307.58738   
std      26.494628     2.923431    11.708825   245.885335   234.17651   
min      10.080000     1.000000     0.508500    10.678500    10.17000   
25%      32.875000     3.000000     5.924875   124.422375   118.49750   
50%      55.230000     5.000000    12.088000   253.848000   241.76000   
75%      77.935000     8.000000    22.445250   471.350250   448.90500   
max      99.960000    10.000000    49.650000  1042.650000   993.00000   

       gross margin percentage  gross income      Rating  
count             1.000000e+03   1000.000000  1000.00000  
mean              4.761905e+00     15.379369     6.97270  
std               6.220360e-14     11.708825     1.71858  
min               4.761905e+00      0.508500     4.00000  
25%               4.761905e+00      5.924875     5.50000  
50%               4.761905e+00     12.088000     7.00000  
75%               4.761905e+00     22.445250     8.50000  
max               4.761905e+00     49.650000    10.00000

# EDA Preparation

What was the best Product line for sales? How much was?

In [54]:
sales_Product_line=df.groupby(['Product line']).sum()
sales_Product_line.rename(columns={'Rating':"Sum of Rating"},inplace=True)
sales_Product_line

Unit price  Quantity     Tax 5%       Total      cogs  \
Product line                                                                    
Electronic accessories     9103.77       971  2587.5015  54337.5315  51750.03   
Fashion accessories       10173.35       902  2585.9950  54305.8950  51719.90   
Food and beverages         9745.54       952  2673.5640  56144.8440  53471.28   
Health and beauty          8337.88       854  2342.5590  49193.7390  46851.18   
Home and lifestyle         8850.71       911  2564.8530  53861.9130  51297.06   
Sports and travel          9460.88       920  2624.8965  55122.8265  52497.93   

                        gross margin percentage  gross income  Sum of Rating  
Product line                                                                  
Electronic accessories               809.523810     2587.5015         1177.2  
Fashion accessories                  847.619048     2585.9950         1251.2  
Food and beverages                   828.571429     2673.5640         1237.7  
Health and beauty                    723.809524     2342.5590         1064.5  
Home and lifestyle                   761.904762     2564.8530         1094.0  
Sports and travel                    790.476190     2624.8965         1148.1

#best product line is Food and beverages,total sale = 56144.8440

#What was the best Branch for sales? How much was?

In [55]:
Branch_sales=df.groupby(['Branch']).sum() 
Branch_sales

Unit price  Quantity     Tax 5%        Total       cogs  \
Branch                                                            
A         18625.49      1859  5057.1605  106200.3705  101143.21   
B         18478.88      1820  5057.0320  106197.6720  101140.64   
C         18567.76      1831  5265.1765  110568.7065  105303.53   

        gross margin percentage  gross income  Rating  
Branch                                                 
A                   1619.047619     5057.1605  2389.2  
B                   1580.952381     5057.0320  2263.6  
C                   1561.904762     5265.1765  2319.9

the highest sales for Branch  is 'C' and sales are = 110568.7065

# To finds payment methods

In [56]:
sales_Payment=df.groupby(['Payment']).sum()
sales_Payment.head()

Unit price  Quantity    Tax 5%       Total       cogs  \
Payment                                                              
Cash           19525.09      1896  5343.170  112206.570  106863.40   
Credit card    16916.68      1722  4798.432  100767.072   95968.64   
Ewallet        19230.36      1892  5237.767  109993.107  104755.34   

             gross margin percentage  gross income  Rating  
Payment                                                     
Cash                     1638.095238      5343.170  2397.7  
Credit card              1480.952381      4798.432  2178.0  
Ewallet                  1642.857143      5237.767  2397.0

Cash payment are higher

In [57]:
pd.value_counts(df['Product line'])

Fashion accessories       178
Food and beverages        174
Electronic accessories    170
Sports and travel         166
Home and lifestyle        160
Health and beauty         152
Name: Product line, dtype: int64

In [58]:
pd.value_counts(df['Payment'])

Ewallet        345
Cash           344
Credit card    311
Name: Payment, dtype: int64

In [59]:
#adding month column
df['Month'] = df['Date'].str[0:1]
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  \
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1   
1  10:29         Cash   76.40                 4.761905        3.8200     9.6   
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4   
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4   
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3   

  Month  
0     1  
1     3  
2     3  
3     1  
4     2

In [60]:
#to find sales by months
sales_Month=df.groupby(['Month']).sum()
sales_Month

Unit price  Quantity    Tax 5%       Total       cogs  \
Month                                                          
1        19753.89      1965  5537.708  116291.868  110754.16   
2        17159.52      1654  4629.494   97219.374   92589.88   
3        18758.72      1891  5212.167  109455.507  104243.34   

       gross margin percentage  gross income  Rating  
Month                                                 
1                  1676.190476      5537.708  2470.2  
2                  1442.857143      4629.494  2142.6  
3                  1642.857143      5212.167  2359.9

In [61]:
Month_Branch = df.groupby(['Month','Branch']).sum()
Month_Branch

Unit price  Quantity     Tax 5%       Total      cogs  \
Month Branch                                                          
1     A          6349.53       685  1841.9585  38681.1285  36839.17   
      B          6544.41       600  1770.2885  37176.0585  35405.77   
      C          6859.95       680  1925.4610  40434.6810  38509.22   
2     A          5430.93       493  1421.9105  29860.1205  28438.21   
      B          5919.62       624  1639.2510  34424.2710  32785.02   
      C          5808.97       537  1568.3325  32934.9825  31366.65   
3     A          6845.03       681  1793.2915  37659.1215  35865.83   
      B          6014.85       596  1647.4925  34597.3425  32949.85   
      C          5898.84       614  1771.3830  37199.0430  35427.66   

              gross margin percentage  gross income  Rating  
Month Branch                                                 
1     A                    566.666667     1841.9585   842.3  
      B                    528.571429     1770.2885   755.0  
      C                    580.952381     1925.4610   872.9  
2     A                    447.619048     1421.9105   658.7  
      B                    519.047619     1639.2510   763.9  
      C                    476.190476     1568.3325   720.0  
3     A                    604.761905     1793.2915   888.2  
      B                    533.333333     1647.4925   744.7  
      C                    504.761905     1771.3830   727.0

In [62]:
Month_Branch_Product_line= df.groupby(['Month','Branch','Product line']).sum()
Month_Branch_Product_line

Unit price  Quantity    Tax 5%  \
Month Branch Product line                                             
1     A      Electronic accessories     1107.00       117  304.8225   
             Fashion accessories        1121.44       114  326.0710   
             Food and beverages          750.40        90  221.2490   
             Health and beauty           663.07        70  188.6950   
             Home and lifestyle         1513.28       175  491.1235   
             Sports and travel          1194.34       119  309.9975   
      B      Electronic accessories      953.99       119  319.0370   
             Fashion accessories        1231.53       103  291.0760   
             Food and beverages         1235.44       102  314.7275   
             Health and beauty           979.56        91  304.7565   
             Home and lifestyle          800.23        73  218.4020   
             Sports and travel          1343.66       112  322.2895   
      C      Electronic accessories      889.22        97  272.8685   
             Fashion accessories        1190.68       119  304.0490   
             Food and beverages         1251.54       133  395.9535   
             Health and beauty          1215.18        93  286.6995   
             Home and lifestyle         1023.58        94  266.4145   
             Sports and travel          1289.75       144  399.4760   
2     A      Electronic accessories     1004.17        89  247.7510   
             Fashion accessories         913.92        79  246.3635   
             Food and beverages         1158.39       127  335.9155   
             Health and beauty           649.66        55  138.8325   
             Home and lifestyle          822.64        68  227.2205   
             Sports and travel           882.15        75  225.8275   
      B      Electronic accessories     1016.41       118  318.3930   
             Fashion accessories        1329.55       118  292.2435   
             Food and beverages          884.06       107  264.5150   
             Health and beauty           869.81       102  278.8775   
             Home and lifestyle          743.15        83  221.8975   
             Sports and travel          1076.64        96  263.3245   
      C      Electronic accessories      918.36       106  260.6610   
             Fashion accessories        1473.42        98  366.6245   
             Food and beverages         1325.54       115  351.9675   
             Health and beauty           966.87       109  277.6355   
             Home and lifestyle          534.29        54  142.9955   
             Sports and travel           590.49        55  168.4485   
3     A      Electronic accessories     1181.10       116  319.6700   
             Fashion accessories         854.83        70  205.3040   
             Food and beverages         1279.73        96  260.1260   
             Health and beauty          1030.80       132  272.3655   
             Home and lifestyle         1294.05       128  349.1415   
             Sports and travel          1204.52       139  386.6845   
      B      Electronic accessories      771.58        79  174.5435   
             Fashion accessories         839.24        76  198.2670   
             Food and beverages          657.50        61  145.2760   
             Health and beauty          1234.47       127  367.8260   
             Home and lifestyle         1232.32       139  395.3750   
             Sports and travel          1279.74       114  366.2050   
      C      Electronic accessories     1261.94       130  369.7550   
             Fashion accessories        1218.74       125  355.9965   
             Food and beverages         1202.94       121  383.8340   
             Health and beauty           728.46        75  226.8710   
             Home and lifestyle          887.17        97  252.2830   
             Sports and travel           599.59        66  182.6435   

                                          Total     cogs  \


In [63]:
#to sales of gender
sales_Branch=df.groupby(['Gender']).sum()
sales_Branch.head()

Unit price  Quantity    Tax 5%       Total       cogs  \
Gender                                                          
Female    27687.24      2869  7994.425  167882.925  159888.50   
Male      27984.89      2641  7384.944  155083.824  147698.88   

        gross margin percentage  gross income  Rating  
Gender                                                 
Female              2385.714286      7994.425  3489.2  
Male                2376.190476      7384.944  3483.5

#high sales by Females

In [64]:
#To find avg rating of product line
Product_line= df[['Product line']]
Rating = df[['Rating']]
df_final = pd.concat([Product_line,Rating],axis=1)
print(df_final.head())
print(df_final.shape)

             Product line  Rating
0       Health and beauty     9.1
1  Electronic accessories     9.6
2      Home and lifestyle     7.4
3       Health and beauty     8.4
4       Sports and travel     5.3
(1000, 2)


In [68]:
Avg_Rating = df_final.groupby(['Product line']).mean()
Avg_Rating

Rating
Product line                    
Electronic accessories  6.924706
Fashion accessories     7.029213
Food and beverages      7.113218
Health and beauty       7.003289
Home and lifestyle      6.837500
Sports and travel       6.916265

In [34]:
df_final2= pd.concat([sales_Product_line,Avg_Rating],axis=1)
df_final2

Unit price  Quantity     Tax 5%       Total      cogs  \
Product line                                                                    
Electronic accessories     9103.77       971  2587.5015  54337.5315  51750.03   
Fashion accessories       10173.35       902  2585.9950  54305.8950  51719.90   
Food and beverages         9745.54       952  2673.5640  56144.8440  53471.28   
Health and beauty          8337.88       854  2342.5590  49193.7390  46851.18   
Home and lifestyle         8850.71       911  2564.8530  53861.9130  51297.06   
Sports and travel          9460.88       920  2624.8965  55122.8265  52497.93   

                        gross margin percentage  gross income  Rating  \
Product line                                                            
Electronic accessories               809.523810     2587.5015  1177.2   
Fashion accessories                  847.619048     2585.9950  1251.2   
Food and beverages                   828.571429     2673.5640  1237.7   
Health and beauty                    723.809524     2342.5590  1064.5   
Home and lifestyle                   761.904762     2564.8530  1094.0   
Sports and travel                    790.476190     2624.8965  1148.1   

                          Rating  
Product line                      
Electronic accessories  6.924706  
Fashion accessories     7.029213  
Food and beverages      7.113218  
Health and beauty       7.003289  
Home and lifestyle      6.837500  
Sports and travel       6.916265

the highest avg rating of product line is Food and beverages = 7.113218

In [35]:
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  \
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1   
1  10:29         Cash   76.40                 4.761905        3.8200     9.6   
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4   
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4   
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3   

  Month  
0     1  
1     3  
2     3  
3     1  
4     2

In [70]:
df.drop(["Invoice ID","City"],axis=1,inplace=True) # to hide label
df.head()

Branch Customer type  Gender            Product line  Unit price  Quantity  \
0      A        Member  Female       Health and beauty       74.69         7   
1      C        Normal  Female  Electronic accessories       15.28         5   
2      A        Normal    Male      Home and lifestyle       46.33         7   
3      A        Member    Male       Health and beauty       58.22         8   
4      A        Normal    Male       Sports and travel       86.31         7   

    Tax 5%     Total       Date   Time      Payment    cogs  \
0  26.1415  548.9715   1/5/2019  13:08      Ewallet  522.83   
1   3.8200   80.2200   3/8/2019  10:29         Cash   76.40   
2  16.2155  340.5255   3/3/2019  13:23  Credit card  324.31   
3  23.2880  489.0480  1/27/2019  20:33      Ewallet  465.76   
4  30.2085  634.3785   2/8/2019  10:37      Ewallet  604.17   

   gross margin percentage  gross income  Rating Month  
0                 4.761905       26.1415     9.1     1  
1                 4.761905        3.8200     9.6     3  
2                 4.761905       16.2155     7.4     3  
3                 4.761905       23.2880     8.4     1  
4                 4.761905       30.2085     5.3     2

In [71]:
df_num = df.select_dtypes(include=[np.number])# to seperated number data from main data
df_num.head()

Unit price  Quantity   Tax 5%     Total    cogs  gross margin percentage  \
0       74.69         7  26.1415  548.9715  522.83                 4.761905   
1       15.28         5   3.8200   80.2200   76.40                 4.761905   
2       46.33         7  16.2155  340.5255  324.31                 4.761905   
3       58.22         8  23.2880  489.0480  465.76                 4.761905   
4       86.31         7  30.2085  634.3785  604.17                 4.761905   

   gross income  Rating  
0       26.1415     9.1  
1        3.8200     9.6  
2       16.2155     7.4  
3       23.2880     8.4  
4       30.2085     5.3

In [72]:
df_obj = df.select_dtypes(include=[np.object]) # to seperated charctsristic (objected) data from main data
df_obj.head()

Branch Customer type  Gender            Product line       Date   Time  \
0      A        Member  Female       Health and beauty   1/5/2019  13:08   
1      C        Normal  Female  Electronic accessories   3/8/2019  10:29   
2      A        Normal    Male      Home and lifestyle   3/3/2019  13:23   
3      A        Member    Male       Health and beauty  1/27/2019  20:33   
4      A        Normal    Male       Sports and travel   2/8/2019  10:37   

       Payment Month  
0      Ewallet     1  
1         Cash     3  
2  Credit card     3  
3      Ewallet     1  
4      Ewallet     2

In [43]:
dummies = pd.get_dummies(df[["Gender"]])
dummies

Gender_Female  Gender_Male
0                1            0
1                1            0
2                0            1
3                0            1
4                0            1
..             ...          ...
995              0            1
996              1            0
997              0            1
998              0            1
999              1            0

[1000 rows x 2 columns]